modules/boltlib/spec/functions/run_plan_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe 'run_plan' do
+  before(:each) do
+    Puppet[:tasks] = true
+  end
+
+  context "when invoked" do
+    context 'can be called as' do
+      it 'run_plan(name) referencing a plan defined in the manifest' do
+        pending('wip')
```

Comment:
@hlindberg I couldn't figure out a good way to write this test within the structure of rspec-puppet. Any ideas?

Reply:
This test may make more sense to keep with the `plan` implementation, and have a dummy function to check that it can be used.

Reply:
Updated to use Pal, but the test fails
```
  1) run_plan when invoked can be called as run_plan(name) referencing a plan defined in the manifest
     # wip
     Failure/Error:
       compiler.evaluate_string(<<-CODE)
       plan run_me() { "worked1" }
       $a = run_plan('run_me')
       notice $a
       CODE

     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Function Call, Function run_plan(): Unknown plan: 'run_me' at line 2:18 on node michaels-mbp.corp.puppetlabs.net
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/loader/../../../../lib/puppet/functions/run_plan.rb:36:in `run_plan'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/dispatch.rb:60:in `invoke'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/dispatcher.rb:43:in `block in dispatch'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/dispatcher.rb:42:in `catch'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/dispatcher.rb:42:in `dispatch'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/function.rb:46:in `block in call'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/function.rb:45:in `catch'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/functions/function.rb:45:in `call'
     # :2:in `block in call_function'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/runtime3_support.rb:307:in `eval'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/runtime3_support.rb:307:in `block in call_function'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/util/profiler/around_profiler.rb:58:in `profile'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/util/profiler.rb:51:in `profile'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/runtime3_support.rb:305:in `call_function'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:963:in `call_function_with_block'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:932:in `eval_CallNamedFunctionExpression'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:48:in `block in visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `each'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:92:in `visit_this_1'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:81:in `evaluate'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:370:in `eval_AssignmentExpression'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:48:in `block in visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `each'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:92:in `visit_this_1'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:81:in `evaluate'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:660:in `block in eval_BlockExpression'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:660:in `each'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:660:in `reduce'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:660:in `eval_BlockExpression'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:48:in `block in visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `each'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `visit_this'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:92:in `visit_this_1'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:81:in `evaluate'
     # :in `stack'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/puppet_stack.rb:30:in `eval'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/puppet_stack.rb:30:in `stack'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:737:in `eval_Program'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:90:in `visit_this_1'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/evaluator/evaluator_impl.rb:81:in `evaluate'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/parser/evaluating_parser.rb:63:in `evaluate'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/parser/evaluating_parser.rb:45:in `evaluate_string'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:87:in `evaluate_string'
     # ./spec/functions/run_plan_spec.rb:16:in `block (6 levels) in <top (required)>'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:516:in `block (2 levels) in main'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/parser/script_compiler.rb:47:in `block in compile'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/context.rb:65:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet.rb:263:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/parser/script_compiler.rb:42:in `compile'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:514:in `block in main'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/context.rb:65:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet.rb:263:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:483:in `main'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:228:in `block in with_script_compiler'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/context.rb:65:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet.rb:263:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:225:in `with_script_compiler'
     # ./spec/functions/run_plan_spec.rb:15:in `block (5 levels) in <top (required)>'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:411:in `block in in_environment_context'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/context.rb:65:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet.rb:263:in `override'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:403:in `in_environment_context'
     # /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet_pal.rb:303:in `in_tmp_environment'
     # ./spec/functions/run_plan_spec.rb:14:in `block (4 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # ArgumentError:
     #   Function run_plan(): Unknown plan: 'run_me'
     #   /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/loader/../../../../lib/puppet/functions/run_plan.rb:36:in `run_plan'
```

Reply:
Found a problem in the PAL API - PR is up here against the feature branch: https://github.com/puppetlabs/puppet/pull/6373